### PR TITLE
Adding support for block_suggestion handlers

### PR DESCRIPTION
**Pull request by selfcontained**
###  Summary

This adds support for routing a `block_suggestion` payload to an optional `blockSuggestion` handler provided by the function module. It follows the same patterns as `block_actions`.

## Testing
To test this, you'll need an app with function interactivity setup. You'll want to include an external select in a block-kit message/modal view that looks like so:

```json
{
  "block_id": "topics",
  "type": "section",
  "text": { "type": "mrkdwn", "text": "Select the meeting topics" },
  "accessory": {
    "action_id": "topics-action",
    "type": "external_select",
    "min_query_length": 1,
    "placeholder": {
      "type": "plain_text",
      "text": "Select",
      "emoji": true,
    },
  },
}
```

You can then export a plain `blockSuggestion` handler from your function file to return options:

```ts
// @ts-ignore - testing out handler w/o types
export const blockSuggestion = ({ inputs, body, token, env, team_id }) => {
  return {
    "options": [
      {
        "text": {
          "type": "plain_text",
          "text": "Option 1",
        },
        "value": "value-1",
      },
      {
        "text": {
          "type": "plain_text",
          "text": "Option 2",
        },
        "value": "value-2",
      },
      {
        "text": {
          "type": "plain_text",
          "text": "Option 3",
        },
        "value": "value-3",
      },
    ],
  };
};
```

This should render a select dropdown that is populated with your returned options. Just like normal external selects, you can access what the user has typed via `body.value` if you wanted to use it as part of a query for the options.

image

**Note - when an option is selected, your app receives a block_actions payload for that selection that you should add a handler for. That can look something like this to just ack the selection:**
```
.addBlockActionsHandler("topics-action", ({ body, action }) => {
    console.log("topics block action", action);
    return {};
  })
```

* Local Mode testing can be done by adding the following hook entry to your `slack.json` file:

```
"start": "deno run -q --config=deno.jsonc --allow-read --allow-net https://raw.githubusercontent.com/slackapi/deno-slack-runtime/191ac86700c98c0f7b97be4b46a25bf1d6b493cc/src/local-run.ts"
```

* Deployed testing - This requires a dev environment that's running this version of the runtime - sync up w/ @selfcontained for details on that.

### Requirements (place an `x` in each `[ ]`)

* [x] I've read and understood the [Contributing Guidelines](https://github.com/{project_slug}/blob/main/.github/CONTRIBUTING.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

File: src/dispatch-payload.ts [hunk 1]
```diff
@@ -78,7 +86,7 @@ export const DispatchPayload = async (
         break;
       default:
         throw new UnhandledEventError(
-          `Received a ${eventType} payload but there is no matching handler for the ${functionCallbackId} function.`,
+          `Received an unsupported event of type: "${eventType}" for the ${functionCallbackId} function.`,
```
Comment (selfcontained) [inline src/dispatch-payload.ts:89]: Updating the message here to better reflect this condition.

File: src/dispatch-payload.ts [hunk 2]
```diff
@@ -111,15 +119,11 @@ function getFunctionCallbackID(
     case EventTypes.FUNCTION_EXECUTED:
       return (payload as InvocationPayload<FunctionInvocationBody>)?.body?.event
         ?.function?.callback_id ?? "";
+    // These all extract the callback_id from the default location
```
Comment (filmaj) [inline src/dispatch-payload.ts:122]: Nice!

File: src/run-block-suggestion.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import { UnhandledEventError } from "./run-unhandled-event.ts";
+
+export const RunBlockSuggestion = async (
+  payload: InvocationPayload<BlockSuggestionInvocationBody>,
+  functionModule: FunctionModule,
+  // deno-lint-ignore no-explicit-any
+): Promise<any> => {
+  const { body, context } = payload;
+  const env = context.variables || {};
+  const team_id = context.team_id || "";
```
Comment (filmaj) [inline src/run-block-suggestion.ts:16]: Maybe worth dropping a TODO in here to check what shape enterprise_id comes in in these events. Hopefully similarly to how it comes in for block actions: https://github.com/slackapi/deno-slack-runtime/pull/38/files#diff-e306b98ca4957a72f121f185cfd25e3084c42300d1467b06a9e31597008e5e8aR17

**Review by filmaj — APPROVED** [review]
LGTM!

File: src/run-block-suggestion.ts (hunk 3 above)
Comment (selfcontained) [reply src/run-block-suggestion.ts:16]: I can take a peek and then add it to this PR so we can merge #38 first, and have it available for these handlers as well.

**Pull request merged**